### PR TITLE
Add Registered Model View

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

* Registered Model View
Screenshot 2024-09-18 at 01 39 06

* Empty View
Screenshot 2024-09-18 at 01 41 32

* Model archival
Screenshot 2024-09-18 at 01 39 21

## How Has This Been Tested?

Added both unit testing and integration testing. As well ass runing the backend in mock mode.

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [X] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work.
- [X] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

If you have UI changes

- [X] The developer has added tests or explained why testing cannot be added.
- [X] Included any necessary screenshots or gifs if it was a UI change.
- [X] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: clients/ui/frontend/src/app/pages/modelRegistry/screens/RegisteredModels/RegisteredModelTableRow.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  const { preferredModelRegistry } = React.useContext(ModelRegistrySelectorContext);
+  const [isArchiveModalOpen, setIsArchiveModalOpen] = React.useState(false);
+  const [isRestoreModalOpen, setIsRestoreModalOpen] = React.useState(false);
+  const rmUrl = registeredModelUrl(rm.id, preferredModelRegistry?.name);
+
+  const actions = [
+    {
+      title: 'View details',
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      onClick: () => {}, // TODO: @Griffin-Sullivan uncomment this once model versions is active ---> navigate(`${rmUrl}/${ModelVersionsTab.DETAILS}`),
```
Comment (Griffin-Sullivan) [inline clients/ui/frontend/src/app/pages/modelRegistry/screens/RegisteredModels/RegisteredModelTableRow.tsx:39]: :saluting_face:

File: clients/ui/frontend/src/app/pages/modelRegistry/screens/components/ArchiveRegisteredModelModal.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import { Flex, FlexItem, Stack, StackItem, TextInput } from '@patternfly/react-core';
+import { Modal } from '@patternfly/react-core/deprecated';
```
Comment (Griffin-Sullivan) [inline clients/ui/frontend/src/app/pages/modelRegistry/screens/components/ArchiveRegisteredModelModal.tsx:3]: I'm also hitting this in my current PR that's not up yet. Will we want to update all of our Modals in a future sprint?

**Review by Griffin-Sullivan — APPROVED** [review]
I've tested this quite a bit already since my current PR is branched from here. Everything looks good to me.

**Review by ederign — APPROVED** [review]

**Comment by ederign** [comment]
@lucferbux can you please rebase this?

**Pull request merged**